runner/src/server/services/uploadService.ts

```diff
@@ -238,9 +238,17 @@ export class UploadService {
     if (request.pre?.warning && page?.controller === "UploadPageController") {
       const update = Object.entries(originalFilenames).reduce(
```

Comment:
does it need to be a loop/reduce? iirc, `originalFileNames` should store a map of all files uploaded in this session. 

if _this_ page has a section, and lets say we've already got a few things in `originalFileNames` at the end of the reducer we'll end up with just one file. 

e.g.

```
const originalFileNames = {
  "a": "a.csv",
  "b": "b.csv"
  "c": "c.csv"
}

const page = {
  section: "details"
}

const update = Object.entries(originalFileNames).reduce((acc, [key, value]) => {

    if (page?.section) {
      return {
        ...acc,
        [page.section]: {
          [key]: value.location
        }
      };
    }
    return {
      ...acc,
      [key]: value.location
    };
  }, {});

console.log(update)  // { details: { c: undefined } }

```

I _think_ we might be safe because of how the cache service / hoek manages merging.. but we should just be on the safe side, and only mutate what we need to?